python/langsmith/_internal/otel/_otel_exporter.py

```diff
@@ -332,12 +351,64 @@ def _update_span_for_run(self, op: SerializedRunOperation, run_info: dict) -> No
                     span.end(end_time=end_time_utc_nano)
                 else:
                     span.end()
-                # Remove the span from our dictionary
+                # Remove the span and timestamp from our dictionaries
                 del self._spans[op.id]
```

Comment:
Read an article this morning about how `delete` in JS is slow, not sure about Python

Maybe good to combine and have both use one method?